### PR TITLE
recovery: use old dc recovery only on ENXIO errors

### PR DESCRIPTION
**Pull request by abudnik**

### CODE REVIEW

File: tests/pytests/isolated/test_isolated.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
```
Comment (shaitan) [inline tests/pytests/isolated/test_isolated.py]: `tests/pytests/isolated` directory is useless, please, move `test_isolated.py` into `tests/pytests/`.

File: tests/pytests/test_recovery.py [hunk 2]
```diff
@@ -1112,16 +1107,7 @@ def cleanup_backends():
...
-        remove_all_blobs(scope.session)
-        enable_backends(scope.session, scope.routes.addresses_with_backends())
-
-        # wait for route-list updates
-        counter = 0
-        while scope.routes.addresses_with_backends() != scope.session.routes.addresses_with_backends():
-            assert counter <= 5  # 5 seconds should be enough to receive and handle all route-list updates
-            time.sleep(0.5)
-            counter += 0.5
+        cleanup_backends(scope.session, scope.session.routes.addresses_with_backends(), scope.routes.addresses_with_backends())
```
Comment (shaitan) [inline tests/pytests/test_recovery.py]: Please, fix new PEP8 errors/warnings.

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -296,7 +298,8 @@ add_custom_command(TARGET test_python
         pip install pytest-timeout &&
         pip install msgpack-python &&
         export LD_LIBRARY_PATH=.:.. &&
-        env ${TEST_ENV} py.test ${PYTESTS_FLAGS} ${CMAKE_CURRENT_SOURCE_DIR}/pytests
+        env ${TEST_ENV} py.test ${PYTESTS_FLAGS} ${CMAKE_CURRENT_SOURCE_DIR}/pytests &&
```
Comment (abudnik) [inline tests/CMakeLists.txt]: No, this is a bad idea.

File: tests/pytests/isolated/test_isolated.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+@pytest.mark.incremental
+class TestIsolatedRecovery:
+    namespace = "TestIsolatedRecovery"
+
+    @pytest.mark.usefixtures("servers")
+    def test_disable_backends(self, simple_node):
+        '''
+        Leave one unique group for each node.
```
Comment (abudnik) [inline tests/pytests/isolated/test_isolated.py]: Because servers used both in dc and merge recovery tests?

File: tests/test_base.cpp [hunk 5]
```diff
@@ -825,7 +825,7 @@ start_nodes_config::start_nodes_config(std::ostream &debug_stream, const std::ve
 , monitor(true)
 , srw(true)
 , isolated(false)
-, client_node_flags(0)
+, client_node_flags(DNET_CFG_NO_ROUTE_LIST)
```
Comment (abudnik) [inline tests/test_base.cpp]: Is `DNET_CFG_NO_ROUTE_LIST` really needed for **client** node?

File: tests/CMakeLists.txt (hunk 3 above)
Comment (shaitan) [inline tests/CMakeLists.txt]: Why?

File: tests/test_base.cpp (hunk 5 above)
Comment (shaitan) [inline tests/test_base.cpp]: It depends on usage, but how this answers my question?

File: tests/pytests/server.py [hunk 6]
```diff
@@ -21,12 +21,34 @@
 sys.path.insert(0, "")  # for running from cmake
 
 
+def make_backends(backends_count, group_id):
+    '''
+    Returns configuration of a single group that has @backends_count backends.
+    '''
+    backends = []
+    for i in xrange(backends_count):
```
Comment (shaitan) [inline tests/pytests/server.py]: `i` here and `node`[there](https://github.com/interiorem/elliptics/pull/112/files#diff-dcb2340e644831a7140b6d308d00fadfR40) are useless and can be replaced by `_`.

File: tests/pytests/conftest.py [hunk 7]
```diff
@@ -38,6 +38,8 @@ def pytest_addoption(parser):
                      help='Number of nodes that should be run')
     parser.addoption('--recovery-keys', action='store', default=10,
                      help='Number of keys that would be used at test_recovery')
+    parser.addoption('--isolated', action='store_true', default=False,
```
Comment (shaitan) [inline tests/pytests/conftest.py]: This option should be removed and its usage.

File: tests/pytests/test_isolated.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+from test_recovery import recovery, RECOVERY, cleanup_backends, write_data, check_data
+
+
+@pytest.yield_fixture(scope="module")
+def servers(request):
+    servers = [{'backends': make_backends(backends_count=1, group_id=g)} for g in 1, 2]
+    _servers = Servers(without_cocaine=True,
+                       servers=servers,
+                       isolated=True,
+                       path='servers_isolated')
```
Comment (shaitan) [inline tests/pytests/test_isolated.py]: Why not 'isolated_servers'??

Comment (abudnik) [inline tests/pytests/test_isolated.py]: fixed

File: tests/pytests/server.py (hunk 6 above)
Comment (abudnik) [inline tests/pytests/server.py]: fixed

File: tests/pytests/conftest.py (hunk 7 above)
Comment (abudnik) [inline tests/pytests/conftest.py]: fixed

**Pull request merged**